server/src/main/java/io/unitycatalog/server/UnityCatalogServer.java

```diff
@@ -43,68 +46,101 @@
 
 public class UnityCatalogServer {
   private static final Logger LOGGER = LoggerFactory.getLogger(UnityCatalogServer.class);
-
-  private SecurityConfiguration securityConfiguration;
-  private SecurityContext securityContext;
+  private static final String BASE_PATH = "/api/2.1/unity-catalog/";
+  private static final String CONTROL_PATH = "/api/1.0/unity-control/";
+  private final Server server;
+  private final ServerProperties serverProperties;
+  private final SecurityContext securityContext;
 
   static {
     System.setProperty("log4j.configurationFile", "etc/conf/server.log4j2.properties");
     Configurator.initialize(null, "etc/conf/server.log4j2.properties");
   }
 
-  Server server;
-  private static final String basePath = "/api/2.1/unity-catalog/";
-  private static final String controlPath = "/api/1.0/unity-control/";
-
   public UnityCatalogServer() {
-    this(8080);
+    this(new Builder());
   }
 
-  public UnityCatalogServer(int port) {
-
+  private UnityCatalogServer(Builder builder) {
+    setDefaults(builder);
     Path configurationFolder = Path.of("etc", "conf");
+    SecurityConfiguration securityConfiguration = new SecurityConfiguration(configurationFolder);
 
-    securityConfiguration = new SecurityConfiguration(configurationFolder);
-    securityContext =
+    this.securityContext =
         new SecurityContext(configurationFolder, securityConfiguration, "server", INTERNAL);
+    this.serverProperties = builder.serverProperties;
+    this.server = initializeServer(builder);
+  }
+
+  private void setDefaults(Builder builder) {
+    if (builder.port == 0) {
+      builder.port(8080);
+    }
+    if (builder.serverProperties == null) {
+      builder.serverProperties(ServerProperties.getInstance());
+    }
+    if (builder.credentialOperations == null) {
+      AwsCredentialVendor awsCredentialVendor = new AwsCredentialVendor(builder.serverProperties);
+      AzureCredentialVendor azureCredentialVendor =
+          new AzureCredentialVendor(builder.serverProperties);
+      GcpCredentialVendor gcpCredentialVendor = new GcpCredentialVendor(builder.serverProperties);
+      CredentialOperations credentialOperations =
+          new CredentialOperations(awsCredentialVendor, azureCredentialVendor, gcpCredentialVendor);
+      builder.credentialOperations(credentialOperations);
+    }
+  }
 
-    ServerBuilder sb = Server.builder().serviceUnder("/docs", new DocService()).http(port);
-    addServices(sb);
+  private Server initializeServer(Builder unityCatalogServerBuilder) {
```

Comment:
nit: this is named just builder in the constructor. lets be consistent.

Reply:
i see that you are trying to differentiate from armeriaServerBuilder. thats okay then.